From 321da973f0b7e012748ead6482b36d2b3a08aa36 Mon Sep 17 00:00:00 2001
From: Caleb Xu <calebcenter@live.com>
Date: Sat, 18 Aug 2018 18:40:24 +0000
Subject: [PATCH 177/177] makefile: include Polly flags

I don't know if these actually do anything helpful in terms of performance or
energy efficiency.

Change-Id: I8e2223d44868037806d16ba7dcf592162578d88e
---
 Makefile | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index c2df4f018f4b..c5df95676bf9 100644
--- a/Makefile
+++ b/Makefile
@@ -395,7 +395,18 @@ endif
 ifneq ($(CLANG_ENABLE_IA),1)
 CLANG_IA_FLAG	= -no-integrated-as
 endif
-CLANG_FLAGS	:= $(CLANG_TARGET) $(CLANG_GCC_TC) $(CLANG_IA_FLAG) -meabi gnu
+POLLY           := -O3 -mllvm -polly \
+		   -mllvm -polly-ast-use-context \
+		   -mllvm -polly-opt-fusion=max \
+		   -mllvm -polly-opt-maximize-bands=yes \
+		   -mllvm -polly-run-dce \
+		   -mllvm -polly-dependences-computeout=0 \
+		   -mllvm -polly-dependences-analysis-type=value-based \
+		   -mllvm -polly-position=after-loopopt \
+		   -mllvm -polly-run-inliner \
+		   -mllvm -polly-detect-keep-going \
+		   -mllvm -polly-rtc-max-arrays-per-group=40
+CLANG_FLAGS	:= $(CLANG_TARGET) $(CLANG_GCC_TC) $(CLANG_IA_FLAG) $(POLLY) -meabi gnu
 endif
 
 # Use USERINCLUDE when you must reference the UAPI directories only.
-- 
2.18.0

